sdk/include/opentelemetry/sdk/trace/always_off_sampler.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/sampler.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+/**
+ * The always off sampler always returns NOT_RECORD, effectively disabling
+ * tracing functionality.
+ */
+class AlwaysOffSampler : public Sampler
+{
+public:
+  /**
+   * @return Returns NOT_RECORD always
+   */
+  SamplingResult ShouldSample(
+    const SpanContext *parent_context,
+    trace_api::TraceId trace_id,
```

Comment:
There is potential optimization to this, by avoiding the trace_id creation (random number generation) at all. Probably worth creating an issue for tracking.

Reply:
Alright, I'll bring this up to Johannes later today.

Reply:
I don't see how we could easily avoid that. The `trace_id` is created only _once_ for every trace anyway. And even if an always-off sampler is active, we might still need the trace id for an outgoing context.

Also, the spec requires the trace id to be passed to a sampler.